blockchain_connector/common/connector_common/worker_delegate.py

```diff
@@ -101,44 +104,58 @@ def add_update_worker_to_chain(self, wids_in_kv, wids_onchain):
 
             result = None
             if wid in wids_onchain:
-                result = self._worker_instance.worker_update(
-                    worker_id, details)
-                if result != ContractResponse.SUCCESS:
-                    logging.error("Error while updating worker {} \
-                            to blockchain".format(wid))
-                else:
-                    logging.info("Updated worker {} to blockchain".format(wid))
+                if wid not in self._worker_details_map or \
+                        self._worker_details_map[wid] != details:
+                    result = self._worker_instance.worker_update(
+                        worker_id, details)
+                    if result != ContractResponse.SUCCESS:
+                        logging.error("Error while updating worker {}"
+                                      " to blockchain".format(wid))
```

Comment:
You could return if contract response is not success. Otherwise we will end up updating worker status to ACTIVE.